music/src/music_vae/model.ts

```diff
@@ -908,17 +927,44 @@ class MusicVAE {
     if (!this.chordEncoder && chordProgression) {
       throw new Error('Unexpected chord progression provided.');
     }
+    if (this.chordEncoder && this.dataConverter.endTensor &&
+        chordProgression.length > 1) {
+      throw new Error(
+          'Multiple chords not supported when using variable-length segments.');
+    }
 
     if (!this.initialized) {
       await this.initialize();
     }
 
     const numSteps = this.dataConverter.numSteps;
+    const numSegments = this.dataConverter.numSegments;
 
     const ohSeqs: tf.Tensor2D[] = tf.tidy(() => {
-      const controls = this.chordEncoder ?
-          this.chordEncoder.encodeProgression(chordProgression, numSteps) :
-          undefined;
+      let controls: tf.Tensor2D = undefined;
+      if (this.chordEncoder) {
+        // Construct the chord-conditioning tensors. For models where each
+        // segment is a separate track, the same chords should be used for
+        // each segment. If an initial chord is specified by the data
+        // converter, use that chord for the first step only.
+        const numChordSteps = this.dataConverter.SEGMENTED_BY_TRACK ?
+            numSteps / numSegments :
+            numSteps;
+        const encodedChordProgression = this.dataConverter.INITIAL_CHORD ?
```

Comment:
Is INITIAL_CHORD only used when SEGMENTED_BY_TRACK is used? If so, do we need both?